### ISSUE TITLE
Releasing using release branches

### ISSUE THREAD

**Issue by lukeromanowicz** [google-github-actions/release-please-action#687]
### TL;DR

I want to create release branches once I think my release is ready, which would trigger the please-release action to create a pull request with a changelog between my new release branch and main branch.

### Detailed design

In large projects with multiple developers sometimes you want to cut a release from develop branch and polish it before releasing, while the rest of the team starts working on the scope of some future releases. Once I create a release branch from my develop, I would like the release-please action to make a pull request with changes between the release branch and the main branch. Once merged, the code in the main branch would be tagged as the next release.

### Additional information

I think that to achieve this we are missing some additional option like 'current-branch'

**Comment by chingor13** [comment]
This would need to be implemented on the upstream library (https://github.com/googleapis/release-please).

However, we are highly unlikely to implement something like this because release-please's goal is to update release notes, versions, and manage GitHub releases and tags. We cannot support every type of branch management that folks use for their development.
